crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -863,7 +876,90 @@ where
             Some(recpts) => recpts,
             None => return Err(EthApiError::UnknownBlockNumber),
         };
-        build_transaction_receipt_with_block_receipts(tx, meta, receipt, &all_receipts)
+
+        #[cfg(feature = "optimism")]
+        let optimism_block_meta = {
+            let block = self
+                .provider()
+                .block_by_number(meta.block_number)?
+                .ok_or(EthApiError::UnknownBlockNumber)?;
+
+            let block_timestamp = block.timestamp;
```

Comment:
after reading this a bit closer.

I want the entire function duplicated with mutually exclusive features

because this can be optimized for optimism since we need to fetch the block as well which can happen concurrently with the `cache().get_receipts()` call.

this should also be an additional separate function that returns the meta struct